src/migrations/20250213000000-clean-false-status-changes.js

```diff
@@ -0,0 +1,129 @@
+const { prepMigration } = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+      return queryInterface.sequelize.query(`
+        -- Remove extra GoalStatusChange entiries that:
+        --  - were inserted when Goals didn't actually change status
+        --  - are duplicate insertions of the same status change
+        -- Also update Goals that have the incorrect "oldStatus" value.
+        -- This appears to occur most often on the initial status change
+        -- for a Goal when an associated Objective is designated as
+        -- In Progress. Potentially that is also when the Goal was created
+
+        DROP TABLE IF EXISTS marked_changes;
+        CREATE TEMP TABLE marked_changes
+        AS
+        SELECT
+          id gscid,
+          "oldStatus" oldstat,
+          "newStatus" newstat,
+          LAG("newStatus") OVER (
+              PARTITION BY "goalId"
+              ORDER BY "createdAt", id
+          ) real_previous_status,
+          "newStatus" != COALESCE(
+            LAG("newStatus") OVER (
+              PARTITION BY "goalId"
+              ORDER BY "createdAt", id
+            ), 'NULL'
```

Comment:
should null be quoted here?

Reply:
There's a few places where you quote it below, same question applies

Reply:
Agree I don't think 'NULL' as a string has the same value as NULL. But it might not matter if the value you are checking against will never be 'NULL' as a string.

Reply:
That’s intended, yes